add endpoint for fetching historical prices

saipradyu opened the pull request:
add wiremock mappings for testing

saipradyu committed:
add endpoint for fetching historical prices
add wiremock mappings for testing

saipradyu committed:
add test cases for each fuse endpoint

grundygolem2 commented on src/main/resources/application.yml:
```diff
@@ -47,7 +49,8 @@ spring:
       iexBasePath: http://localhost:${wiremock.server.port}/
-
+      iexCloudPath: http://localhost:${wiremock.server.port}/
+      iexCloudToken: pk_398f22558d024d7ca05e79961bfdb3f7
```
Don't commit tokens in git. Remove this, and wire it in from a file that isn't committed

grundygolem2 reviewed (changes requested):
Looks good, one security change to make

saipradyu committed:
use chart by day to fetch historical price by date
hide iex api key and read from environment variable
minor cleanup of iex rest controller test
